### PR TITLE
[IMP] webiste_cf_turnstile: transition from jQuery to vanilla js

### PR DESCRIPTION
**Pull request by shsa-odoo**
Specification:

This commit focuses on transitioning all frontend code from jQuery to Vanilla JavaScript.
The changes specifically target files in the web.assets_frontend.

task-3928123

### CODE REVIEW

File: addons/website_cf_turnstile/static/src/js/turnstile.js [hunk 1]
```diff
@@ -5,51 +5,60 @@ import publicWidget from '@web/legacy/js/public/public_widget';
...
-        },
+    /**
+     * @override
+    */
+    start: function () {
+        const res = this._super(...arguments);
+        this.cleanTurnstile();
+        if (!this.isEditable && !this.el.querySelector(".s_turnstile") && session.turnstile_site_key) {
+            const mode = new URLSearchParams(window.location.search).get("cf") == "show" ? "always" : "interaction-only";
+            const turnstileElement = document.createElement("div");
```
Comment (bso-odoo) [inline addons/website_cf_turnstile/static/src/js/turnstile.js:16]: `turnstileEl` ?

File: addons/website_cf_turnstile/static/src/js/turnstile.js [hunk 2]
```diff
@@ -5,51 +5,60 @@ import publicWidget from '@web/legacy/js/public/public_widget';
...
 
-        /**
-         * Remove potential existing loaded script/token
-         */
-        cleanTurnstile: function () {
-            if (this.$('.s_turnstile').length) {
-                this.$('.s_turnstile').remove();
-            }
-        },
+            const scriptElement1 = document.createElement("script");
```
Comment (bso-odoo) [inline addons/website_cf_turnstile/static/src/js/turnstile.js:24]: `script1El` ?

File: addons/website_cf_turnstile/static/src/js/turnstile.js [hunk 3]
```diff
@@ -5,51 +5,60 @@ import publicWidget from '@web/legacy/js/public/public_widget';
...
 
-        /**
-         * @override
-         * Discard all library changes to reset the state of the Html.
-         */
-        destroy: function () {
-            this.cleanTurnstile();
-            this._super(...arguments);
-        },
+            const scriptElement2 = document.createElement("script");
```
Comment (bso-odoo) [inline addons/website_cf_turnstile/static/src/js/turnstile.js:36]: `script2El` ?

File: addons/website_cf_turnstile/static/src/js/turnstile.js [hunk 4]
```diff
@@ -5,51 +5,60 @@ import publicWidget from '@web/legacy/js/public/public_widget';
...
+            scriptElement2.className = "s_turnstile";
+            scriptElement2.src = "https://challenges.cloudflare.com/turnstile/v0/api.js";
+
+            const formSendElement = this.el.querySelector(".s_website_form_send, .o_website_form_send");
```
Comment (bso-odoo) [inline addons/website_cf_turnstile/static/src/js/turnstile.js:40]: `formSendEl` ?

File: addons/website_cf_turnstile/static/src/js/turnstile.js [hunk 5]
```diff
@@ -5,51 +5,60 @@ import publicWidget from '@web/legacy/js/public/public_widget';
...
+            formSendElement.parentNode.insertBefore(scriptElement2, formSendElement.nextSibling);
+        }
+        return res;
+    },
+
+    /**
+     * Remove potential existing loaded script/token
+    */
+    cleanTurnstile: function () {
+        const turnstileElements = this.el.querySelectorAll(".s_turnstile");
```
Comment (bso-odoo) [inline addons/website_cf_turnstile/static/src/js/turnstile.js:52]: `turnstileEls` ?

**Review by bso-odoo — CHANGES_REQUESTED** [review]
Here are a few comments: variable to rename to mach the `*El`, `*Els` naming.

**Comment by shsa-odoo** [comment]
@bso-odoo done!

**Comment by bso-odoo** [comment]
@robodoo r+

**Pull request closed without merging**